src/realtime_metrics/main.py

```diff
@@ -6,17 +6,168 @@
 from sqlalchemy import create_engine, inspect
 from sqlalchemy.orm import sessionmaker
 
-import pandas as pd
+import numpy
+
+def run_analysis():
+    # get all stop time updates and run compute the different metrics
+    tripUpdates = session.query(TripUpdate).group_by(TripUpdate.trip_id).all()
+
+    stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]] = []
+
+    for tripUpdate in tripUpdates:
+        trip_stop_time_updates = session.query(StopTimeUpdate, TripUpdate).join(TripUpdate).filter(TripUpdate.trip_id == tripUpdate.trip_id).filter(StopTimeUpdate.arrival_time > 0).all()
+        for trip_stop_time_update in trip_stop_time_updates:
+            stop_time_updates.append((trip_stop_time_update.TripUpdate, trip_stop_time_update.StopTimeUpdate))
+
+            # check dict
```

Comment:
```suggestion
            # add stop_time_update to dictionary if not present or update if timestamp of tripUpdate is newer than the one in the dictionary
```
Remove or improve the comment